allauth/socialaccount/providers/edmodo/provider.py

```diff
@@ -32,9 +32,13 @@ def extract_common_fields(self, data):
 
     def extract_extra_data(self, data):
         return dict(
-            user_type=data.get("type"),
-            profile_url=data.get("url"),
             avatar_url=data.get("avatars").get("large"),
+            email=data.get("email"),
```

Comment:
Can we make this? Similar to Cognito, I prefer to store the raw data as is, instead of creating our own abstraction:

``` python
    def extract_extra_data(self, data):
        ret = dict(data)
        # NOTE: For backwards compatibility
        user_type = data.get("type")
        if user_type is not None:
            ret["user_type"] = user_type
        profile_url = data.get("url")
        if profile_url:
            ret["profile_url"] = profile_url
        avatar_url = data.get("avatars", {}).get("large")
        if avatar_url:
            ret["avatar_url"] = avatar_url
        # (end NOTE)
        return ret
```

Reply:
Thanks. I just committed code similar to this. For backwards compatibility, I always set the `user_type`, `profile_url` and `avatar_url` keys, even if their values are `None`.